crates/runtime/src/actor/component/mod.rs

```diff
@@ -730,18 +858,20 @@ impl Instance {
     }
 
     /// Invoke an operation on an [Instance] producing a result.
-    #[instrument(level = "debug", skip(self, params, instance, name), fields(interface = instance, function = name))]
-    pub async fn call(
+    #[instrument(skip(self, instance, name, incoming, outgoing), fields(interface = instance, function = name))]
+    pub async fn call<C>(
         &mut self,
         instance: &str,
         name: &str,
-        params: Vec<wrpc_transport::Value>,
-    ) -> anyhow::Result<Vec<wrpc_transport::Value>> {
+        incoming: Reader,
+        outgoing: SubjectWriter,
```

Comment:
```suggestion
        incoming: C::Incoming,
        outgoing: C::Outgoing,
```

And you'll need to bind `C: wrpc_transport::Invoke`